### PR TITLE
Update Sphinx documentation

### PR DESCRIPTION
**Pull request by xDaile**
The celery was not compatible with Sphinx >=6.2.0 because of this issue (https://github.com/celery/celery/pull/7978). As the new version of celery, compatible with Sphinx > 6.2.0 was released, we need to update the outdated documentation. 

Currently, the Sphinx library has available a newer version 7.0.1 instead of 6.2.1, however, this version is not compatible with the sphinx-rtd-theme as mentioned in  (https://github.com/readthedocs/sphinx_rtd_theme/issues/1463), so we need to stick to 6.2.1 for now.

- Documentation to new modules was added.
- docs/requeirements.txt was reordered and updated with opentelementry packages
- Added workflow job for building the docs in Github CI/CD

Additionally /web/static/api_v1.yaml was updated.

### CODE REVIEW

File: iib/common/tracing.py [hunk 1]
```diff
@@ -7,8 +7,8 @@
 
 Usage:
     @instrument_tracing()
-    def func():
-        pass
+      def func():
+          pass
```
Comment (xDaile) [inline iib/common/tracing.py:11]: This was the only way I found to write this comment, where Sphinx was not throwing errors.

File: docs/requirements.txt [hunk 2]
```diff
@@ -13,7 +13,18 @@ recommonmark
 requests
 ruamel.yaml
 # https://github.com/sphinx-doc/sphinx/issues/2796
-sphinx==6.2.0
+sphinx==6.2.1
 sphinx_rtd_theme
 tenacity
 typing-extensions
+opentelemetry-api
```
Comment (yashvardhannanavati) [inline docs/requirements.txt:20]: Could you please arrange these dependencies alphabetically

File: iib/common/tracing.py (hunk 1 above)
Comment (yashvardhannanavati) [reply iib/common/tracing.py:11]: Could you please share the error you were getting without this extra tab?

Comment (xDaile) [reply iib/common/tracing.py:11]: `/iib/common/tracing.py:docstring of iib.common.tracing:9:Unexpected indentation.`

**Review by yashvardhannanavati — DISMISSED** [review]

File: docs/requirements.txt [hunk 3]
```diff
@@ -1,19 +1,30 @@
...
 dogpile.cache
 flask
 flask-login
 flask-migrate
 flask-sqlalchemy
+opentelemetry-api
+opentelemetry-exporter-otlp
+opentelemetry-instrumentation
+opentelemetry-instrumentation-botocore
+opentelemetry-instrumentation-celery
```
Comment (chandwanitulsi) [inline docs/requirements.txt:13]: why do we have to add opentelemetry libraries for this PR?

File: docs/requirements.txt [hunk 4]
```diff
@@ -1,19 +1,30 @@
-celery
+# https://github.com/sphinx-doc/sphinx/issues/2796
```
Comment (chandwanitulsi) [inline docs/requirements.txt:1]: Is this comment only valid for line 27 `sphinx==6.2.1` ?

File: docs/requirements.txt (hunk 3 above)
Comment (yashvardhannanavati) [reply docs/requirements.txt:13]: I believe this is only for docs. They are already listed as dependencies in setup.py

Comment (xDaile) [reply docs/requirements.txt:13]: Exactly. As I added modules, which included parts of the code working with opentelemetry, into documentation, Sphinx was throwing errors about missing opentelemetry libraries.

File: docs/requirements.txt (hunk 4 above)
Comment (xDaile) [reply docs/requirements.txt]: Sure! Thanks for catch.

File: docs/requirements.txt [hunk 5]
```diff
@@ -1,19 +1,29 @@
...
-python-memcached
 psycopg2-binary
+python-memcached
+python-qpid-proton
 recommonmark
 requests
 ruamel.yaml
-# https://github.com/sphinx-doc/sphinx/issues/2796
-sphinx==6.2.0
+sphinx==6.2.1
```
Comment (lipoja) [inline docs/requirements.txt:26]: May I ask you a favor?
Could you test latest sphinx release - v7.0.1 I would go with pinning latest rather then older one. 
But only if everything passes.

Comment (lipoja) [reply docs/requirements.txt:26]: _# And also dependabot is suggesting to pin latest version in different PR._

Comment (lipoja) [reply docs/requirements.txt:26]: _# And also dependabot is suggesting to pin latest version in different PR._

Comment (xDaile) [reply docs/requirements.txt:26]: I already wrote the reason why this is not possible in the PR description ([sphinx-rtd-theme issue](https://github.com/readthedocs/sphinx_rtd_theme/issues/1463)).

Comment (lipoja) [reply docs/requirements.txt:26]: I should do the review better next time - starting reading description :D ... thanks!

**Review by lipoja — APPROVED** [review]

**Review by lipoja — DISMISSED** [review]

**Review by lipoja — APPROVED** [review]

**Review by yashvardhannanavati — APPROVED** [review]

**Review by chandwanitulsi — APPROVED** [review]

**Pull request merged**